meson.build

```diff
@@ -0,0 +1,320 @@
+project('notcurses', ['c', 'cpp'],
+        default_options:
+          ['c_std=gnu11',
+           'cpp_std=gnu++17',
+           'default_library=both',
+           'c_args=-Wall -Wextra -W -Wshadow -Wformat -fexceptions',
+           'cpp_args=-Wall -Wextra -W -Wshadow -Wformat -fexceptions'
```

Comment:
These should not be here. `-Wall`and `-Wextra` should be enabled with `warning_level=3` instead. The other flags should be added with `add_project_args` instead after testing that they are supported with `get_supported_arguments`.